### ISSUE TITLE
[FR] Create a config to turn on autoreplacement of magrittr pipe with native pipe

### ISSUE THREAD

**Issue by jarodmeng** [R-nvim/R.nvim#57]
This is a feature request.

I'm in the process of migrating all my pipe usage from the `magrittr` `%>%` pipe to the R native `|>` pipe. But old habits truly die hard and I find my self mixing the two pipes all the time in my script. As a result, I have to do global replacement often. I'm wondering if it's possible to enable some kind of auto-replacement of `%>%` to `|>` in real time when typing. There can be a setting which turns on and off this behavior.

Congrats on the new launch and thanks for all the hard work over many years!

**Comment by PMassicotte** [comment]
I am not sure to understand. Do you mean that `|>` would be inserted even if you type `%>%`?

**Comment by jarodmeng** [comment]
> I am not sure to understand. Do you mean that `|>` would be inserted even if you type `%>%`?

That's right. Much like when `_` is typed, it's replaced with `<-` if `R_assign` is turned on.

**Comment by jalvesaq** [comment]
You may put this in your `init.lua`:

```lua
vim.cmd.iabbrev('%>', '|>')
```

or, even better, to unlearn the old habit:

```lua
vim.cmd.iabbrev('%>', 'TRY_AGAIN')
```

You can also create an autocmd to automatically replace "%<%" with "<|" when the buffer is saved.

**Comment by jarodmeng** [comment]
I tried to set up the abbreviation, but it doesn't work. It may have something to do with the characters allowed in abbreviation.

I ended up doing this in 2 steps.
1) I used `grep --include={*.R,*.r,*.qmd,*.rmd} -rli '%>%' * | xargs -i@ sed -i 's/%>%/|>/g' @` to replace all existing `%>%` with `|>` in my folder.
2) I included `autocmd BufWritePre *.R,*.r,*.qmd,*.rmd,*.Qmd,*.Rmd :%s/%>%/|>/ge | write` in my setup to automatically replace `%>%` with `|>` in R files when saving.

It doesn't solve the real-time replacement when typing, but at least help make sure that my R files now always have `|>` instead of `%>%`.

**Comment by PMassicotte** [comment]
I think it was addressed in #116

**Comment by jalvesaq** [comment]
We are not going to add autoreplacement of `%>%` with `|>`....

Hence, I propose that we add this to the documentation:

```diff
diff --git a/doc/R.nvim.txt b/doc/R.nvim.txt
index 14b1512..921a62e 100644
--- a/doc/R.nvim.txt
+++ b/doc/R.nvim.txt
@@ -1185,8 +1185,12 @@ R.nvim is aware of `.Rproj` files (the project-level configuration files used
 by RStudio). This may, for example, change whether `<Plug>RInsertPipe` inserts
 `|>` or `%>%` for a particular project. See the option |rproj_prioritise| for
 a full list of the behaviours which may be affected by `.Rproj` files, or to
-change whether they should affect things at all.
-
+change whether they should affect things at all. You can also use R.nvim
+config option `pipe_version` to choose what kind of pipe operator to insert.
+Valid values are "native" and "magrittr":
+>lua
+   pipe_version = "native"
+<
 
 ------------------------------------------------------------------------------
 6.6. Object Browser and completion options                    *objbr_w*
```

Then, close this issue.
